inference/core/workflows/core_steps/models/foundation/cogvlm.py

```diff
@@ -0,0 +1,344 @@
+            )
+
+    async def run_locally(
+        self,
+        images: Batch[WorkflowImageData],
+        prompt: str,
+        json_output: Optional[Dict[str, str]],
+    ) -> BlockResult:
+        if json_output:
+            prompt = (
+                f"{prompt}\n\nVALID response format is JSON:\n"
+                f"{json.dumps(json_output, indent=4)}"
+            )
+        images_prepared_for_processing = [
+            image.to_inference_format(numpy_preferred=True) for image in images
+        ]
+        raw_output = await get_cogvlm_generations_locally(
+            image=images_prepared_for_processing,
+            prompt=prompt,
+            model_manager=self._model_manager,
+            api_key=self._api_key,
+        )
+        structured_output = turn_raw_lmm_output_into_structured(
+            raw_output=raw_output,
+            expected_output=json_output,
+        )
+        predictions = [
+            {
+                "raw_output": raw["content"],
+                "image": raw["image"],
+                "structured_output": structured,
+                **structured,
+            }
+            for raw, structured in zip(raw_output, structured_output)
+        ]
+        for prediction, image in zip(predictions, images):
+            prediction[PARENT_ID_KEY] = image.parent_metadata.parent_id
+            prediction[ROOT_PARENT_ID_KEY] = (
+                image.workflow_root_ancestor_metadata.parent_id
+            )
+        return predictions
+
+    async def run_remotely(
+        self,
+        images: Batch[WorkflowImageData],
+        prompt: str,
+        json_output: Optional[Dict[str, str]],
+    ) -> BlockResult:
+        if json_output:
+            prompt = (
+                f"{prompt}\n\nVALID response format is JSON:\n"
+                f"{json.dumps(json_output, indent=4)}"
+            )
+        inference_images = [i.to_inference_format() for i in images]
+        raw_output = await get_cogvlm_generations_from_remote_api(
+            image=inference_images,
+            prompt=prompt,
+            api_key=self._api_key,
+        )
+        structured_output = turn_raw_lmm_output_into_structured(
+            raw_output=raw_output,
+            expected_output=json_output,
+        )
+        predictions = [
+            {
+                "raw_output": raw["content"],
+                "image": raw["image"],
+                "structured_output": structured,
+                **structured,
+            }
+            for raw, structured in zip(raw_output, structured_output)
+        ]
+        for prediction, image in zip(predictions, images):
+            prediction[PARENT_ID_KEY] = image.parent_metadata.parent_id
+            prediction[ROOT_PARENT_ID_KEY] = (
+                image.workflow_root_ancestor_metadata.parent_id
+            )
+        return predictions
+
+
+async def get_cogvlm_generations_locally(
+    image: List[dict],
```

Comment:
We are passing multiple images here, is there a reason not to call this `images`?